### DISCUSSION TITLE
Is there a standard way to create a "back button"?

### DISCUSSION THREAD

**Discussion by zedascouves** [Budibase/budibase#11188] [Show and tell]
My app is to be mainly used on a tablet, so I want to create a standard back button. I've searched but I couldn't find anything specific, so I've created the following:

![image](https://github.com/Budibase/budibase/assets/594831/112152eb-d2d1-4c06-a833-fa95e444f8ca)

In the "embed" field:

```
<!DOCTYPE html>
<html>
<body>
<button class="spectrum-Button spectrum-Button--sizeM spectrum-Button--cta svelte-1gv5n3y cd17fc8d794774a44aa5bbe652ad93894-dom" onclick="history.back()">Back</button>
</body>
</html>
```

I've copied the class attribute from another out-of-the-box button.

This works quite well, but I'd like to have an easier and standard way to do it. I did try with the "Navigate to" action and adding the "history.back()" function in the Javascript field, but as soon as I write it, the browser goes back a page! So it seems that the Javascript box is executing the code in real time.

Any other suggestion? Thanks!

**Comment by ConorWebb96** [comment]
Hey @zedascouves,

I'm assuming you are doing this because your users don't have a set path to get to the specific page you are on? (e.g. there a numbers way they can land on this page) If this is the case your method would be the best method in my opinion.

The other option would be to create a feature request, which would add an app action that would take the users previous page in and use this to redirect them back to the previous page they were on. You can do this if you wish here; https://github.com/Budibase/budibase/issues/new/choose 

**Side Notes**

If you have a set structure for your users e.g. they all will be coming from a specific route you could simply use the `Navigate to` action and redirect them that way. This could then be attached to the button component avoiding you having to use the embed component like you are currently doing.

Additionally, if you are using a form and you are directing them down the chain, i'd recommend using steps and containing everything on the one page. The `Change form step` action is very useful in this case, it has a range of different options. E.g. previous, next, first and specific step. In this case the previous step is the one you want to use for going back.

This is quite off topic, but there was a similar but different question asked within the community discord, you can read a bit on this here; https://discord.com/channels/733030666647765003/1126183236037529660/1126183236037529660

I hope this is helpful.

**Comment by zedascouves** [comment]
Hi @ConorWebb96,

Thanks for clearing that up. You assume correctly about the navigation path.

The `form step` suggestion doesn't make sense for this case. I do agree that the best way would be to use the `Navigate to` action, but in a way that would resolve (this may not be the best word...) the `history.back()` function and not a fully formed URL.

In any case, I guess this thread may be relevant for anyone else looking for this kind on functionality.

Cheers!
